tests/tb_gateway_mqtt_client_tests.py

```diff
@@ -13,10 +13,287 @@
 # limitations under the License.
 
 import unittest
+from unittest.mock import MagicMock,patch
 from time import sleep, time
+import threading
+from tb_gateway_mqtt import TBGatewayMqttClient, TBSendMethod, GATEWAY_CLAIMING_TOPIC, GATEWAY_RPC_TOPIC, GATEWAY_MAIN_TOPIC
 
-from tb_gateway_mqtt import TBGatewayMqttClient
 
+class TestGwUnsubscribe(unittest.TestCase):
+    def setUp(self):
+        self.client = TBGatewayMqttClient("localhost", 1883, "dummy_token")
+        if not hasattr(self.client, "_lock"):
+            self.client._lock = threading.Lock()
+        self.client._TBGatewayMqttClient__sub_dict = {
+            "device1|attr1": {1: lambda msg: "callback1"},
+            "device2|attr2": {2: lambda msg: "callback2"},
+        }
+
+    def test_unsubscribe_specific(self):
+        sub_dict = self.client._TBGatewayMqttClient__sub_dict
+        self.assertIn(1, sub_dict["device1|attr1"])
+        self.assertIn(2, sub_dict["device2|attr2"])
+
+        self.client.gw_unsubscribe(1)
+
+        sub_dict = self.client._TBGatewayMqttClient__sub_dict
+        self.assertNotIn(1, sub_dict["device1|attr1"])
+        self.assertIn(2, sub_dict["device2|attr2"])
+
+    def test_unsubscribe_all(self):
+        self.client.gw_unsubscribe('*')
+        self.assertEqual(self.client._TBGatewayMqttClient__sub_dict, {})
+
+class TestGwSendRpcReply(unittest.TestCase):
+    def setUp(self):
+        self.client = TBGatewayMqttClient("localhost", 1883, "dummy_token")
+        if not hasattr(self.client, "_lock"):
```

Comment:
What this check for? As mentioned in another comment - this lock creates during initialisation of TBDeviceMqttClient instance.